src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -293,7 +293,8 @@ def on_save_checkpoint(self) -> Dict:
     def on_load_checkpoint(self, state_dict: Dict) -> None:
         # cache the dataloader state dict until the dataloader objects are available
         self._dataloader_state_dict = state_dict.get("dataloader_state_dict")
-        self._batches_that_stepped = state_dict.get("_batches_that_stepped", 0)
+        # restore global step instead to make sure logging works correctly if checkpoints <v1.6.5 used to resume
+        self._batches_that_stepped = state_dict.get("_batches_that_stepped", self.global_step)
```

Comment:
This could be done without changing the logic if we had checkpoint migration #9396 :)

Reply:
yep.. looks really relevant.